### ISSUE TITLE
feat: introduce spherical Arakawa C-grid operators

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#162]
## Summary

Add spherical coordinate support via a `SphericalArakawaCGrid2D/3D` grid hierarchy and spherical operator classes that reuse the public raw stencil primitives from #161.

**Depends on**: #161 (raw stencil extraction)

## Motivation

Currently, spherical operators exist as 12 standalone functions in `geographic.py` with no grid integration — every call requires manually passing `cos_lat_T`, `cos_lat_U`, `cos_lat_V`, `cos_lat_X`, `dlon`, `dlat`, `R`. This is error-prone and inconsistent with the class-based Cartesian API.

The goal is to provide a spherical grid that inherits from the Cartesian grid, with operator classes that reuse the same raw stencils (Layer 1) but apply spherical metric scaling (Layer 2) and compose spherical compound operators (Layer 3).

### Grid hierarchy

```
ArakawaCGrid2D                          (Cartesian, scalar dx/dy)
  └── SphericalArakawaCGrid2D           (inherits, adds metric terms)
        - dlon, dlat: float             (radians)
        - R: float                      (planet radius, default R_EARTH)
        - cos_lat_T: Array [Ny, Nx]     (cos(latitude) at T-points)
        - cos_lat_U: Array [Ny, Nx]     (cos(latitude) at U-points)
        - cos_lat_V: Array [Ny, Nx]     (cos(latitude) at V-points)
        - cos_lat_X: Array [Ny, Nx]     (cos(latitude) at X-points)
        - lat_T, lon_T: Array           (coordinates for diagnostics/plotting)

ArakawaCGrid3D
  └── SphericalArakawaCGrid3D           (inherits, adds metric terms)
        - horizontal_grid() → SphericalArakawaCGrid2D
```

### Factory

```python
SphericalArakawaCGrid2D.from_interior(
    nx_interior=128, ny_interior=64,
    lon_range=(0.0, 360.0),    # degrees
    lat_range=(-80.0, 80.0),   # degrees
    R=R_EARTH,
)
```

The factory computes `dlon`/`dlat` in radians, builds lat/lon coordinate arrays with ghost cells, and precomputes `cos(lat)` at each staggered location (T, U, V, X) with appropriate half-cell offsets.

The parent `dx`/`dy` are set to `R * dlon` / `R * dlat` for backward compatibility with code that reads those attributes.

### Operator classes

Each spherical operator reuses raw stencils from `stencils.py` (#161) with spherical metric scaling:

```python
class SphericalDifference2D(eqx.Module):
    grid: SphericalArakawaCGrid2D

    def diff_lon_T_to_U(self, h):
        """dh/dλ at U-points: (h[j,i+1] - h[j,i]) / (R·cos(lat_U)·dlon)"""
        raw = diff_x_fwd(h)  # Layer 1: raw stencil from stencils.py
        metric = self.grid.R * self.grid.cos_lat_U[1:-1, 1:-1] * self.grid.dlon
        return interior(_safe_div(raw, metric), h)
```

**Key design decisions**:
- **Interpolation is coordinate-independent** — `Interpolation2D` works as-is with a spherical grid (no wrapper needed)
- **Coriolis is purely algebraic** — existing `Coriolis2D` works with a spherical grid
- **Compound operators compose differently** — metric terms appear *inside* derivatives:
  - Spherical divergence: `1/(R·cosφ) · [du/dλ + d(v·cosφ)/dφ]` (not just Cartesian div with different dx/dy)
  - Spherical curl: `1/(R·cosφ) · [dv/dλ - d(u·cosφ)/dφ]`
- **Pole handling**: `_safe_div(a, b)` returns NaN where `|cos(lat)| < 1e-12`

### Cos(lat) staggering detail

```
lat_T[j] = lat_min + (j - 1) * dlat        (ghost at j=0, j=Ny-1)
lat_U[j] = lat_T[j]                         (same latitude as T)
lat_V[j] = lat_T[j] + 0.5 * dlat           (half-cell north)
lat_X[j] = lat_V[j]                         (same latitude as V)

cos_lat_*[j, :] = cos(lat_*[j])            broadcast over longitude
```

### Phase 1: Spherical grid classes
- `SphericalArakawaCGrid2D(ArakawaCGrid2D)` with factory, cos_lat arrays, coordinate arrays
- `SphericalArakawaCGrid3D(ArakawaCGrid3D)` with `horizontal_grid()` override
- Tests: factory shapes, cos_lat values, staggering offsets
- Export from `__init__.py`

### Phase 2: Spherical difference primitives
- `SphericalDifference2D` — 6 primitive + 2 second-order methods using raw stencils
- `SphericalDifference3D` (vmap + `zero_z_ghosts`)
- Cross-validated against existing `geographic.py` functions (bitwise match)

### Phase 3: Spherical interpolation
- Verify `Interpolation2D(grid=SphericalArakawaCGrid2D(...))` works directly
- Document decision: no separate wrapper needed

### Phase 4: Spherical compound operators
- `SphericalDivergence2D` — `1/(R·cosφ) · [du/dλ + d(v·cosφ)/dφ]`
- `SphericalVorticity2D` — relative vorticity, potential vorticity, PV fluxes
- `SphericalLaplacian2D` — zonal + meridional terms
- `SphericalCoriolis2D` (or document that `Coriolis2D` works as-is)
- Geostrophic velocity
- 3D variants (vmap + `zero_z_ghosts`)
- All validated against existing `geographic.py` tests

### Phase 5: Spherical diffusion and advection (stretch goal)
- `SphericalDiffusion2D/3D` — flux-form with spherical metrics
- Spherical advection — metric-scaled flux divergence

### Phase 6: Deprecate `geographic.py` standalone functions
- Add deprecation warnings
- Migration guide in docs
- Remove after one release cycle

## Public API additions

```python
# Grid classes
"SphericalArakawaCGrid2D", "SphericalArakawaCGrid3D",

# Operators
"SphericalDifference2D", "SphericalDifference3D",
"SphericalDivergence2D",
"SphericalVorticity2D", "SphericalVorticity3D",
"SphericalLaplacian2D",
"SphericalCoriolis2D", "SphericalCoriolis3D",  # if needed
"SphericalDiffusion2D", "SphericalDiffusion3D",
```

## Testing strategy

1. **Unit tests**: Each spherical primitive tested against hand-computed values on simple fields (constant, linear, sinusoidal)
2. **Cross-validation**: Spherical operator output must match existing `geographic.py` functions identically (bitwise)
3. **Equatorial limit**: At the equator (`cos(lat)=1`), spherical operators should reduce to Cartesian operators (with `dx=R·dlon`, `dy=R·dlat`)
4. **Pole tests**: Verify NaN at poles, no inf, no crashes
5. **JIT compatibility**: All operators must work under `jax.jit`
6. **Gradient compatibility**: All operators must work under `jax.grad`

## Future extensions (not in scope)

- **Cylindrical coordinates**: Same 3-layer pattern with `dr`, `r·dθ`, `dz` metrics
- **Rectilinear (stretched) grids**: 1D array `dx(i)` — raw stencils unchanged, only metric scaling changes
- **Curvilinear orthogonal grids**: 2D arrays `h₁(i,j)·dξ`, `h₂(i,j)·dη` — same raw stencils, 2D metric arrays
- **Beta-plane / f-plane approximations**: Coordinate-independent, works with any grid
